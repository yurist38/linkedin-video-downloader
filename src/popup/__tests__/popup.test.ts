import { shallowMount, mount, VueWrapper, config } from '@vue/test-utils';
import Popup from '../popup.vue';

config.global.components = {
  'it-button': {
    template: '<div>it-button</div>',
  },
  'it-divider': {
    template: '<div>it-divider</div>',
  },
};

describe('popup/popup', () => {
  let wrapper: VueWrapper;

  beforeAll(() => {
    chrome.runtime.openOptionsPage = jest.fn();
    chrome.tabs.create = jest.fn();
  });

  it('Should render correctly', () => {
    wrapper = mount(Popup);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('Should open settings page', async () => {
    wrapper = mount(Popup);

    await wrapper.find('#settings-btn').trigger('click');
    expect(chrome.runtime.openOptionsPage).toHaveBeenCalledTimes(1);
  });

  it('Should open donate page', async () => {
    wrapper = mount(Popup);

    await wrapper.find('#donate-btn').trigger('click');
    expect(chrome.tabs.create).toHaveBeenCalledTimes(1);
    expect(chrome.tabs.create).toHaveBeenCalledWith({ url: 'https://www.buymeacoffee.com/roomjs' });
  });
});
