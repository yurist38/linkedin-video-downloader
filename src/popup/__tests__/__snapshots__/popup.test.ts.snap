// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup/popup Should render correctly 1`] = `
<section
  class="content"
>
  <h1>
    LinkedIn Video Downloader
  </h1>
  <p>
    The extension is running in background. In order to use it just go to LinkedIn website and look for a download icon in the corner of every video (you can change the corner in the settings).
  </p>
  <div>
    it-divider
  </div>
  <p>
    <i>
      This is a free 
      <a
        href="https://github.com/yurist38/linkedin-video-downloader"
        target="_blank"
      >
        open source
      </a>
       project. You can support it 
      <a
        href="https://www.buymeacoffee.com/roomjs"
        target="_blank"
      >
        here
      </a>
      . Every donation is appreciated. Thank you!
    </i>
  </p>
  <div
    class="btn-container"
  >
    <div
      id="settings-btn"
      outlined=""
      type="black"
    >
      it-button
    </div>
    <div
      id="donate-btn"
      outlined=""
      type="primary"
    >
      it-button
    </div>
  </div>
</section>
`;
